### PR TITLE
nixos: rework guide

### PR DESCRIPTION
**Pull request by soopyc**
This PR includes a near-complete rewrite of the NixOS guide. For illustrative purposes, `pymdownx.tabbed` and the linked tabs feature was added.

For context, the current NixOS guide is not exactly clear in some places, is outdated and contains code snippets which does not work on the current release. This PR aims to improve readability of the guide and update code snippets.

A deployment of the up-to-date render can be found here: https://t2wiki.soopy.moe

### CODE REVIEW

File: docs/distributions/nixos/installation.md [hunk 1]
```diff
@@ -1,44 +1,215 @@
-# NixOS Installation
+# NixOS Installation Guide
 
-!!! note "Pre-Installation Guide"
-    This guide assumes that you have followed the [pre-installation guide](https://wiki.t2linux.org/guides/preinstall).
+You will need:
 
-1. Partition your disk using `cfdisk` or the tool of your preference, initialize the partitions with the `mkfs` command of the filesystem you want (`mkswap` is for swap) and mount them under `/mnt`.
+- a Mac device with a T2 chip
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/installation.md:5]: Please fix grammar. A new sentence starts with a capital letter

File: docs/distributions/nixos/installation.md [hunk 2]
```diff
@@ -1,44 +1,215 @@
...
+- a spare USB stick or other reasonably large storage medium (>2G recommended)
+- some experience with the command line
+- sufficient mental energy to find and fix errors
 
-    **Note**: You might want to leave a little part of your disk as a FAT32 partition to be able to transfer files easily between MacOS and Linux.
+## Installation Steps
+### Preparation
+Follow the [pre-installation guide](../../guides/preinstall.md). You may not be able to complete some steps without doing so, like [Partitioning](#partitioning).
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/installation.md:12]: Pre-Installation (again, caps)

File: docs/distributions/nixos/installation.md [hunk 3]
```diff
@@ -1,44 +1,215 @@
...
+### Partitioning
+After booting to the live environment, partition the disk (usually `/dev/nvme0n1`) using `cfdisk` or other tools of your preference.
+If you have followed the pre-installation guide, there should be another partition created just for Linux.
 
-    ```sh
-    sudo mkdir -p /lib/firmware/brcm
-    sudo cp /mnt/boot/firmware/* /lib/firmware/brcm
-    sudo modprobe -r brcmfmac && sudo modprobe brcmfmac
-    ```
+- Remove that and allocate new partitions accordingly - there are no hard rules. Just make sure to not delete the EFI partition. See the [NixOS installation manual](https://nixos.org/manual/nixos/stable/#sec-installation-manual-partitioning-UEFI) for examples.
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/installation.md:18]: Make sure to not delete the macOS partition too

File: docs/distributions/nixos/installation.md [hunk 4]
```diff
@@ -1,44 +1,215 @@
...
-4. Edit `/mnt/etc/nixos/configuration.nix`:
-    * Add `"${builtins.fetchGit { url = "https://github.com/NixOS/nixos-hardware.git"; }}/apple/t2"` to `imports`.
-    * Copy the WiFi/Bluetooth firmware to `/mnt/etc/nixos/firmware/brcm` and add this snippet to your `configuration.nix`:
+	This is just an example and does not have to be done this way.
 
-        ```nix
-        hardware.firmware = [
-          (pkgs.stdenvNoCC.mkDerivation {
-            name = "brcm-firmware";
+!!! tip
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/installation.md:39]: Is !!! tip valid?

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

File: docs/distributions/nixos/installation.md (hunk 4 above)
Comment (soopyc) [reply docs/distributions/nixos/installation.md:39]: `!!! tip` is valid per [mkdocs-material's admonitions reference](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#+type:tip).

File: docs/distributions/nixos/faq.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# The livecd does not boot/only shows a blank screen
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/faq.md:1]: LiveCD

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

**Comment by soopyc** [comment]
Most lint errors that could be fixed have been fixed. I could not figure out how to fix `MD046/code-block-style` without breaking content tabs.

File: docs/distributions/nixos/installation.md [hunk 6]
```diff
@@ -1,44 +1,230 @@
...
-
-            buildCommand = ''
-              dir="$out/lib/firmware"
-              mkdir -p "$dir"
-              cp -r ${./files/firmware}/* "$dir"
-            '';
-          })
-        ];
+        ```console
+        $ nixos-rebuild build .#
```
Comment (AdityaGarg8) [inline docs/distributions/nixos/installation.md:127]: Avoid using $ when asking to run a command. It makes things difficult to copy.

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]
A sample of the webpage would be needed to merge this. Probably use GitHub pages on your fork.

**Comment by AdityaGarg8** [comment]
> > A sample of the webpage would be needed to merge this. Probably use GitHub pages on your fork.
> 
> A deployment of the forked wiki is available [here](https://t2wiki.soopy.moe/distributions/nixos/installation) with CF pages.

LGTM

**Comment by sharpenedblade** [comment]
LGTM

File: docs/distributions/nixos/installation.md [hunk 7]
```diff
@@ -1,44 +1,229 @@
...
+Follow the [Pre-Installation guide](../../guides/preinstall.md). You may not be able to complete some steps without doing so, like [Partitioning](#partitioning).
+
+### Partitioning
+
+After booting to the live environment, partition the disk (usually `/dev/nvme0n1`) using `cfdisk` or other tools of your preference.
+If you have followed the pre-installation guide, there should be another partition created just for Linux.
+
+- Remove that and allocate new partitions accordingly - there are no hard rules. Just make sure to not delete the EFI partition and the macOS partition. See the [NixOS Installation Manual](https://nixos.org/manual/nixos/stable/#sec-installation-manual-partitioning-UEFI) for examples.
+- Format the newly provisioned partitions with `mkfs.<filesystem>` commands and `mkswap` a swap partition (if any).
+- Mount the partitions under `/mnt`, and create directories as necessary. Then, enable the swap partition by using `swapon`. You should finally have something that looks similar to this:
```
Comment (Redecorating) [inline docs/distributions/nixos/installation.md:26]: it's been a while since I've used nix, do you have to chroot/nixos-enter before you can enable swap?

**Review by Redecorating — APPROVED** [review]
been a while since i've used nix but this seems good.

File: docs/distributions/nixos/installation.md (hunk 7 above)
Comment (soopyc) [reply docs/distributions/nixos/installation.md:26]: nope - it's the same as other distros.

Comment (AdityaGarg8) [reply docs/distributions/nixos/installation.md:26]: @Redecorating , please approve if you are fine with this PR.

Comment (Redecorating) [reply docs/distributions/nixos/installation.md:26]: its good

**Pull request merged**